jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/auth/AuthConfig.java

```diff
@@ -35,34 +37,47 @@
 @EqualsAndHashCode
 public class AuthConfig {
 
-  public static final AuthConfig EMPTY_AUTH_CONFIG = new AuthConfig("", "", "", "");
+  public static final AuthConfig EMPTY_AUTH_CONFIG = new AuthConfig("", "", "", "", "");
 
   private final String username;
   private final String password;
   private final String email;
   private final String auth;
+  private final String identityToken;
 
   private String authEncoded;
 
   @Builder
-  public AuthConfig(String username, String password, String email, String auth) {
+  public AuthConfig(String username, String password, String email, String auth, String identityToken) {
     this.username = username;
     this.password = password;
     this.email = email;
     this.auth = auth;
-    authEncoded = createAuthEncoded();
+    this.identityToken = identityToken;
+  }
+
+  public AuthConfig(String username, String password, String email, String auth) {
+    this(username, password, email, auth, null);
   }
 
-  public String toHeaderValue() {
-    return authEncoded;
+  public String toHeaderValue(KitLogger logger) {
+    return createAuthEncoded(logger);
```

Comment:
Maybe the nested call to createAuthEncoded is no longer needed (what's the need for the extra-complexity?).

Just move the content of the createAuthEncoded method and remove it.